controllers/argocd_metrics_controller.go

```diff
@@ -274,6 +282,42 @@ func (r *ArgoCDMetricsReconciler) createServiceMonitorIfAbsent(namespace string,
 	return err
 }
 
+func (r *ArgoCDMetricsReconciler) reconcileOperatorMetricsServiceMonitor(reqLogger logr.Logger) error {
+
+	data, err := os.ReadFile(operatorPodNamespacePath)
+	if err != nil {
+		reqLogger.Error(err, "Error retrieving operator's running namespace")
+		return err
+	}
+
+	operatorNS := string(data)
+	desiredMetricsServerName := operatorMetricsServiceName + "." + operatorNS + ".svc"
+
+	existingServiceMonitor := &monitoringv1.ServiceMonitor{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: operatorMetricsMonitorName, Namespace: operatorNS}, existingServiceMonitor)
+
+	if err == nil {
+		changed := false
+
+		if existingServiceMonitor.Spec.Endpoints[0].TLSConfig.ServerName != desiredMetricsServerName {
+			existingServiceMonitor.Spec.Endpoints[0].TLSConfig.ServerName = desiredMetricsServerName
```

Comment:
If there are no items in the `Endpoints` array, We need to create a new `TLSConfig` and add that element isn't it ? Would it not rather better to override the entire desired Endpoint ?
```existingServiceMonitor.Spec.Endpoints = Endpoint[]{&TLSConfig{ServerName: desiredMetricsServerName}}```
Also can there be a scenario where there could be multiple endpoints ?

Reply:
technically yes, I think. having multiple end points would just mean this service monitor is watching multiple services. In our case that won't happen because this service monitor only watches a single metrics service, so there will only be one endpoint 

This PR's scope was restricted to just the changes needed to address a known issue that we want to fix in 1.11, that's why the change is rather specific to just reconciling that one serverName field 

if you think it would be better to just reconcile the entire servicemonitor I am open to that as well
at the moment I don't expect any other fields to get modified though